backend/functions/lib/use-cases/case.assignment.ts

```diff
@@ -21,12 +21,41 @@ export class CaseAssignmentUseCase {
   }
 
   public async createTrialAttorneyAssignments(
-    applicationContext: ApplicationContext,
+    context: ApplicationContext,
     caseId: string,
     newAssignments: string[],
     role: string,
   ): Promise<AttorneyAssignmentResponseInterface> {
-    applicationContext.logger.info(MODULE_NAME, 'New assignments:', newAssignments);
+    const assignmentIds = await this.assignTrialAttorneys(context, caseId, newAssignments, role);
+
+    // Reassign all child cases if this is a joint administration lead case.
+    const consolidationReferences = await this.casesRepository.getConsolidation(context, caseId);
+    const childCaseIds = consolidationReferences
+      .filter(
+        (reference) =>
+          reference.documentType === 'CONSOLIDATION_FROM' &&
+          reference.consolidationType === 'administrative',
+      )
+      .map((reference) => reference.otherCase.caseId);
+    for (const childCaseId of childCaseIds) {
+      await this.assignTrialAttorneys(context, childCaseId, newAssignments, role);
+    }
+
+    return {
+      success: true,
+      message: 'Trial attorney assignments created.',
+      count: assignmentIds.length,
+      body: assignmentIds,
+    };
+  }
+
+  private async assignTrialAttorneys(
```

Comment:
On line 77 - 101 (not a change in this PR), it looks like we are performing the exact same loop 2 times, but just doing something different based on the same filter process.  I'd like to understand if there is reasoning behind that and I'm wondering if there is a more efficient way to handle this.